File: services/teams-service/Models/TeamApplication.cs
namespace TeamsService.Models
{
    public class TeamApplication : BaseModel
    {
        public int UserId { get; set; }
        public int VacancyId { get; set; }
        public string? Text { get; set; }
        public string Status { get; set; } = "Sent";
        public TeamVacancy? TeamVacancy { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: services/main-service/app/Models/TeamApplication.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class TeamApplication extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'vacancy_id',
        'team_id',
        'text',
        'status'
    ];

    /**
     * @return BelongsTo
     */
    public function vacancy(): BelongsTo
    {
        return $this->belongsTo(TeamVacancy::class, 'vacancy_id');
    }

    /**
     * @return BelongsTo
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * @return BelongsTo
     */
    public function team(): BelongsTo
    {
        return $this->belongsTo(Team::class);
    }
}
